shop/src/main/java/ro/msg/learning/shop/controller/ProductController.java

```diff
@@ -0,0 +1,40 @@
+package ro.msg.learning.shop.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.dto.ProductDto;
+import ro.msg.learning.shop.exception.LocationNotFoundException;
+import ro.msg.learning.shop.exception.ProductCategoryNotFoundException;
+import ro.msg.learning.shop.exception.ProductNotFoundException;
+import ro.msg.learning.shop.service.ProductService;
+
+import java.util.List;
+import java.util.UUID;
+
+@RequestMapping("/product")
+@Validated
+@RestController
+public class ProductController {
+    @Autowired
+   private ProductService productService;
+
+    @PostMapping("/new")
+    public ResponseEntity<ProductDto> createProduct(@RequestBody @Validated ProductDto productDto,@RequestParam String productCategoryName,@RequestParam String locationName,@RequestParam Integer quantity) throws ProductCategoryNotFoundException, LocationNotFoundException {
```

Comment:
Mixing `@RequestBody` with '@RequestParam' is not a good idea, especially here. So you decided from business perspective to also check if a ProductCategory exists already in the table and if not to add this. But we should not send the information with '@RequestParam'. We should have that details in the `@RequestBody` as well. That is why DTOs do not have the same structure as the models, to do stuff like this.
You also have a hint here in the requirements:
![image](https://github.com/VladMiron4/Spring-training/assets/138880600/de921380-5b22-4982-a4f4-de876cb96e68)